### ISSUE TITLE
fleet prepare db fails with a db connection error to the wrong address

### ISSUE THREAD

**Issue by benjamin-hering** [kolide/fleet#1727]
### What version of `fleet` are you using (`fleet version --full`)?
 - version 1.0.6
  branch: 	master
  revision: 	45165aa29aae52346ed1c458f19b89947d315c61
  build date: 	2017-12-04T22:50:27Z
  build user: 	marpaia
  go version: 	go1.9.2

### What operating system are you using?

CentOS7

### What did you do?

I successfully setup a quick PoC of fleet to work with using a local MySQL on the same machine as the fleet binary. Now that it's functional, I wanted to build it out with an external MySQL instance. I spun up a fresh database and tried to prepare the db via

`./fleet prepare db --mysql_address=<external mysql address>:3306 --mysql_database=<database> --mysql_username=<username> --mysql_password=<password>` 

I tried this both setting an IP address and a DNS name for the mysql_address flag.

### What did you expect to see?

Migrations complete or something similar to the sample output https://github.com/kolide/fleet/blob/master/docs/infrastructure/fleet-on-centos.md#running-the-fleet-server

### What did you see instead?

`Error creating db connection: Error 1045: Access denied for user '<username>'@'<ip of machine with the fleet binary>' (using password: YES)`

In short, it seems it tried to connect to the box that the fleet binary was on, and not the IP address that I had passed on via the flag for the remote MySQL server, and failed as there was no longer a local running instance of MySQL

**Comment by zwass** [comment]
@benjamin-hering What do you get when you replace `prepare db` with `config_dump`? This should show what configuration values Fleet is using after it has parsed any config files, env vars and flags. The flags should take precedence, but I think this will be a good first debugging step.

This may be easier to debug by joining us in #kolide on [osquery Slack](https://osquery-slack.herokuapp.com/).

**Comment by benjamin-hering** [comment]
Thanks for the response @zwass. I was able to find my issue, and it was completely my own stupidity. I was provisioned a MySQL instance that required SSL, but wasn't attempting to connect with it via anything more than username and password. The shortness of the error details threw me off, but ultimately it was my own mistake. I'll close the issue.
